numpy/f2py/crackfortran.py

```diff
@@ -947,7 +959,17 @@ def analyzeline(m, case, line):
             block = 'python module'
         elif re.match(r'abstract\s*interface', block, re.I):
             block = 'abstract interface'
-        name, args, result, bind = _resolvenameargspattern(m.group('after'))
+        if block == 'type':
+            name, attributes, params = _resolvetypedefpattern(m.group('after'))
+            for att in attributes:
+                if att == 'public' or att == 'private':
+                    groupcache[groupcounter]['vars'][name] = {'attrspec': [att, ]}
+                else:
+                    outmess('analyzeline: ignored type definition attribute "%s"\n' % att.replace('@',''))
```

Comment:
There is no reason to ignore non-access-type attributes, and we need these anyway when implementing derived type support. So, replace the for-loop with
```python
groupcache[groupcounter]['vars'][name] = dict(attrspec = attributes)
```

Reply:
Indeed, much better like that.